### PR TITLE
add error handling

### PR DESCRIPTION
**Pull request by lopesrb**

### CODE REVIEW

File: lib/active_campaign_crm/connection.rb [hunk 1]
```diff
@@ -16,30 +17,60 @@ def get(url, params = {})
...
-      @connection.delete do |req|
+      response = @connection.delete do |req|
         req.url url
         req.headers['Api-Token'] = ActiveCampaignCrm.configuration.api_key
       end
+      handle_response(response)
+    end
+
+    def handle_response(response)
+      return JSON.parse response.body if response.success?
```
Comment (jpgbarbosa) [inline lib/active_campaign_crm/connection.rb:50]: are you using a linter for this? if you're not you should. I'd prefer to wrap `response-body` in `()`

**Review by jpgbarbosa — APPROVED** [review]
That's a cool first approach. You might want to do some sort of std.err and dump a few more info on the gem side. Not sure if worth it tbh. 

What about tests? are you planning to add them?

File: lib/active_campaign_crm/connection.rb (hunk 1 above)
Comment (lopesrb) [reply lib/active_campaign_crm/connection.rb:50]: changed

**Comment by lopesrb** [comment]
@jpgbarbosa yes, I need to test this carefully. but I am leaving this to the end because there is stuff that still can be changed and I don't want to rewrite tests

**Pull request merged**